internal/worker/removal/worker.go

```diff
@@ -62,8 +88,75 @@ func NewWorker(config Config) (worker.Worker, error) {
 }
 
 func (w *removalWorker) loop() (err error) {
-	<-w.catacomb.Dying()
-	return w.catacomb.Err()
+	watch, err := w.cfg.RemovalService.WatchRemovals()
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	if err := w.catacomb.Add(watch); err != nil {
+		return errors.Capture(err)
+	}
+
+	ctx := w.catacomb.Context(context.Background())
+
+	timer := w.cfg.Clock.NewTimer(jobCheckMaxInterval)
+	defer timer.Stop()
+
+	for {
+		select {
+		case <-w.catacomb.Dying():
+			return w.catacomb.ErrDying()
+		case jobIDs := <-watch.Changes():
+			w.cfg.Logger.Infof(ctx, "got removal job changes: %v", jobIDs)
+			if err := w.processRemovalJobs(ctx); err != nil {
+				return errors.Capture(err)
+			}
+			timer.Reset(jobCheckMaxInterval)
+		case <-timer.Chan():
+			if err := w.processRemovalJobs(ctx); err != nil {
+				return errors.Capture(err)
+			}
+			timer.Reset(jobCheckMaxInterval)
+		}
+	}
+}
+
+// processRemovalJobs interrogates *all* current jobs scheduled for removal.
+// For each one whose scheduled start time has passed, we check to see if there
+// is an entry in our runner for it. If there is, it is already being processed
+// and we ignore it. Otherwise, it is commenced in a new runner.
+// This is safe due to the following conditions:
+// - This is the only method interrogating/adding workers to the runner.
+// - It is only invoked from cases in the main event loop, so is Goroutine safe.
+func (w *removalWorker) processRemovalJobs(ctx context.Context) error {
+	jobs, err := w.cfg.RemovalService.GetAllJobs(ctx)
```

Comment:
Can we shuffle the `jobs` slice? Since each job already starts in a separate goroutine.

I'd just rather it be clear that there is no order. It will help weed out poorly written jobs.

Reply:
Note that the only error able to crash the top level worker is the one for adding workers to the runner. This error is not an error that can be caused by the runner or the start closure; it is _only_ about the runner infrastructure.

Given this behaviour and the fact that the runner configuration regards _no_ errors as being fatal, order or not doesn't affect anything. Randomness is inherent in the fact that we don't `ORDER` the results from the jobs query, but I think any comments around this add noise rather than clarity.